tests/integration/test_acm.py

```diff
@@ -53,12 +57,70 @@ def test_import_certificate(self, aws_client):
-        certificate_arn = acm_request_certificate()
+        certificate_arn = acm_request_certificate()["CertificateArn"]
         result = aws_client.acm.describe_certificate(CertificateArn=certificate_arn)
         options = result["Certificate"]["DomainValidationOptions"]
         assert len(options) == 1
 
-    def test_boto_wait_for_certificate_validation(self, acm_request_certificate, aws_client):
-        certificate_arn = acm_request_certificate()
+    def test_boto_wait_for_certificate_validation(
+        self, acm_request_certificate, aws_client, monkeypatch
+    ):
+        monkeypatch.setattr(moto_settings, "ACM_VALIDATION_WAIT", 1)
+        certificate_arn = acm_request_certificate()["CertificateArn"]
         waiter = aws_client.acm.get_waiter("certificate_validated")
-        waiter.wait(CertificateArn=certificate_arn, WaiterConfig={"Delay": 0, "MaxAttempts": 1})
+        waiter.wait(CertificateArn=certificate_arn, WaiterConfig={"Delay": 0.5, "MaxAttempts": 3})
+
+    @pytest.mark.aws_validated
+    @pytest.mark.skip_snapshot_verify(paths=["$..Certificate.SignatureAlgorithm"])
+    def test_certificate_for_subdomain_wildcard(
+        self, acm_request_certificate, aws_client, snapshot, monkeypatch
+    ):
+        snapshot.add_transformer(snapshot.transform.key_value("OID"))
+        snapshot.add_transformer(snapshot.transform.key_value("Serial"))
+        monkeypatch.setattr(moto_settings, "ACM_VALIDATION_WAIT", 2)
+
+        # request certificate for subdomain
+        domain_name = f"test-domain-{short_uid()}.localhost.localstack.cloud"
+        subdomain_pattern = f"*.{domain_name}"
+        create_response = acm_request_certificate(
+            ValidationMethod="DNS", DomainName=subdomain_pattern
+        )
+        cert_arn = create_response["CertificateArn"]
+
+        snapshot.add_transformer(snapshot.transform.regex(domain_name, "<domain-name>"))
+        cert_id = cert_arn.split("certificate/")[-1]
+        snapshot.add_transformer(snapshot.transform.regex(cert_id, "<cert-id>"))
+        snapshot.match("request-cert", create_response)
+
+        def _get_cert_with_records():
+            response = aws_client.acm.describe_certificate(CertificateArn=cert_arn)
+            assert response["Certificate"]["DomainValidationOptions"][0]["ResourceRecord"]
+            return response
+
+        # wait for cert with ResourceRecord CNAME entry
+        response = retry(_get_cert_with_records, sleep=1, retries=30)
+        dns_options = response["Certificate"]["DomainValidationOptions"][0]["ResourceRecord"]
+        snapshot.add_transformer(
+            snapshot.transform.regex(dns_options["Name"].split(".")[0], "<record-prefix>")
+        )
+        snapshot.add_transformer(snapshot.transform.regex(dns_options["Value"], "<record-value>"))
+        snapshot.match("describe-cert", response)
+
+        if is_aws_cloud():
+            # wait until DNS entry has been added (needs to be done manually!)
+            input()
```

Comment:
I think this might be a bit dangerous. If we (sometime in the future) want to run the AWS tests automatically to re-verify, would this block or fail (due to a missing tty)?
What exactly needs to be done here?
/cc @dominikschubert

Reply:
True, 👍  should have been a bit more specific. What needs to be done here is essentially to go to our DNS config (on gandi.net), and add a CNAME record that AWS generates and requires for domain verification (the CNAME record can be retrieved from the AWS console, or is also present in the `DomainValidationOptions.ResourceRecord` in the DescribeCertificate API response).

Good point about tests requiring TTY - at least we could introduce a test marker that indicates that this test requires manual intervention (which, essentially, will be the case for probably all ACM integration tests if we want to parity test them).